pupil-spa/src/app/services/storage/storage.service.ts

```diff
@@ -1,38 +1,33 @@
 import { Injectable } from '@angular/core';
-
-export type StorageKey = 'answers' | 'inputs' | 'session' |
-  'audit' | 'questions' | 'config' | 'pupil' | 'school' | 'access_token' |
-  'feedback' | 'checkstate' | 'device' | 'pending_submission' | 'completed_submission' |
-  'access_arrangements' | 'tokens' | 'time_out' | 'check_start_time';
+import { StorageKeyTypesAll, StorageKeyPrefix } from './storageKey';
 
 @Injectable()
 export class StorageService {
 
-  setItem(key: StorageKey, value: Object | Array<Object>): void {
-
+  setItem(key: StorageKeyTypesAll, value: Object | Array<Object>): void {
```

Comment:
this undefined/null check suggests we need to tighten the settings in `tsconfig.json`